core/templates/core/map_page.html

```diff
@@ -0,0 +1,207 @@
+        }
+        var longerLength = longer.length;
+        if (longerLength == 0) {
+          return 1.0;
+        }
+        return (longerLength - editDistance(longer, shorter)) / parseFloat(longerLength);
+      }
+
+      function editDistance(s1, s2) {
+          s1 = s1.toLowerCase();
+          s2 = s2.toLowerCase();
+
+          var costs = new Array();
+          for (var i = 0; i <= s1.length; i++) {
+            var lastValue = i;
+            for (var j = 0; j <= s2.length; j++) {
+              if (i == 0)
+                costs[j] = j;
+              else {
+                if (j > 0) {
+                  var newValue = costs[j - 1];
+                  if (s1.charAt(i - 1) != s2.charAt(j - 1))
+                    newValue = Math.min(Math.min(newValue, lastValue),
+                      costs[j]) + 1;
+                  costs[j - 1] = lastValue;
+                  lastValue = newValue;
+                }
+              }
+            }
+            if (i > 0)
+              costs[s2.length] = lastValue;
+          }
+          return costs[s2.length];
+        }
+
+      // Get values for map locations
+      window.names = [
+      {% for Location in page.places %}
+          `{{Location.value.name}}`,
+      {% endfor %}
+      ]
+      window.addresses = [
+      {% for Location in page.places %}
+          `{{Location.value.address}}`,
+      {% endfor %}
+      ]
+      window.descriptions = [
+      {% for Location in page.places %}
+          `{{Location.value.description}}`,
+      {% endfor %}
+      ]
+
+      // Parse addresses
+      window.parsedaddresses = []
+      for(i = 0; i< window.addresses.length;i++){
+        var addressString = window.addresses[i];
+        var index = addressString.search("p");
+        var index2 = addressString.search("/p");
+        var addressString = addressString.slice(index + 2,index2-1);
+        window.parsedaddresses.push(addressString);
+      }
+
+      // Initialize and add the map
+      function initMap() {
+        var customMapType = new google.maps.StyledMapType([
+            {
+              elementType: 'labels',
+              stylers: [{visibility: 'off'}]
+            },{
+              featureType: "road",
+              stylers: [{visibility: 'on'}]
+            }
+          ], {
+            name: 'Custom Style'
+        });
+
+
+      var customMapTypeId = 'custom_style';
+      // The location of Troy
+      var Troy = {lat: 42.728, lng: -73.691};
+      // The map, centered at Troy
+      var map = new google.maps.Map(
+          document.getElementById('map'), {zoom: 16, center: Troy});
+
+      // Get Lat and Long from addresses
+      var geocoder = new google.maps.Geocoder();
+
+      // create markers per address
+      for(i=0;i<window.parsedaddresses.length;i++){
+        console.log(window.parsedaddresses[i]);
+        console.log(i);
+        geocoder.geocode( { 'address': window.parsedaddresses[i]}, function(results, status) {
```

Comment:
I'm afraid geocoding the addresses on the frontend for all clients will cause us to hit a rate limit rather quickly once deployed. Perhaps we should request the latitude and longitude on map creation/update and store those lat/lng values then